src/main/java/it/gov/pagopa/cgn/portal/service/DiscountService.java

```diff
@@ -52,6 +52,7 @@ public CrudDiscountWrapper createDiscount(String agreementId, DiscountEntity dis
         ProfileEntity profileEntity = validateDiscount(agreementId, discountEntity, true);
         DiscountEntity toReturn = discountRepository.save(discountEntity);
         if (DiscountCodeTypeEnum.BUCKET.equals(profileEntity.getDiscountCodeType())) {
+            bucketService.createEmptyDiscountBucketCodeSummary(toReturn);
```

Comment:
what if a job starts immediately after this?

Reply:
Right! In the job we have to consider only summaries that have availableCodes > 0. In this case we would sent an expired notification right after the discount creation.

Reply:
Check will be skipped for 0 available codes in https://github.com/pagopa/cgn-onboarding-portal-backend/pull/107/commits/9e9779ea784ca535665776493e2eb2b14a7bb3b3